tensorflow/core/common_runtime/eager/mkl_eager_op_rewrite.cc

```diff
@@ -237,9 +229,10 @@ bool MklEagerOpRewrite::SlowCheckIfKernelRegistered(string op_name,
 }
 
 Status MklEagerOpRewrite::RewriteToMklOp(
-    EagerOperation* orig_op, std::unique_ptr<EagerOperation>* mkl_op,
-    const int op_idx) {
-  mkl_eager_ops_[op_idx].CreateMklOp(orig_op, mkl_op);
+    EagerOperation* orig_op, std::unique_ptr<EagerOperation>* mkl_op) {
+  // TODO(intel-tf): mkl_eager_ops_ lookup can be reduced from twice
+  // (once each in ShouldRewriteOp & RewriteToMklOp) to just once.
```

Comment:
I was thinking about this too. Thank you for adding the TODO!

Reply:
my pleasure!